app/controllers/staff/program_sessions_controller.rb

```diff
@@ -17,4 +17,69 @@ def show
     @program_session = @event.program_sessions.find(params[:id])
     @speakers = @program_session.speakers
   end
+
+  def edit
+    @program_session = @event.program_sessions.find(params[:id])
+    authorize @program_session
+  end
+
+  def update
+    @program_session = @event.program_sessions.find(params[:id])
+    authorize @program_session
+    if @program_session.update(program_session_params)
+      flash[:success] = "#{@program_session.title} was successfully updated."
+      redirect_to event_staff_program_session_path(@event, @program_session)
+    else
+      flash[:danger] = "There was a problem updating this program session."
+      render :edit
+    end
+
+  end
+
+  def new
+    @program_session = @event.program_sessions.new
+    authorize @program_session
+    @speaker = @program_session.speakers.build
+  end
+
+  def create
+    @program_session = @event.program_sessions.build(program_session_params)
+    authorize @program_session
+    @program_session.update(state: ProgramSession::INACTIVE)
+    if @program_session.save
+      redirect_to event_staff_program_session_path(@event,  @program_session)
+      flash[:info] = "Program session was successfully created."
+    else
+      flash[:danger] = "Program session was unable to be saved: #{@program_session.errors.full_messages.to_sentence}"
+      render :new
+    end
+  end
+
+  def update_state
```

Comment:
Let's have state be updated normally via the edit/update form-action interaction.  Please remove all references to this separate action/process to update state.